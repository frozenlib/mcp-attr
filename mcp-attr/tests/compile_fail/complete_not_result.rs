use mcp_attr::{server::{mcp_server, McpServer, RequestContext}};

struct TestServer;

async fn complete_not_result(_value: &str, _cx: &RequestContext) -> Vec<String> {
    vec!["test".to_string()]
}

#[mcp_server]
impl McpServer for TestServer {
    #[prompt]
    async fn test_prompt(&self, #[complete(complete_not_result)] msg: String) -> mcp_attr::Result<String> {
        Ok(msg)
    }
}

fn main() {}