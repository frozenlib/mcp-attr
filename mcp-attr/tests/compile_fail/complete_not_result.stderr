error[E0277]: the `?` operator can only be applied to values that implement `Try`
 --> tests/compile_fail/complete_not_result.rs:6:10
  |
6 | async fn complete_not_result(_value: &str, _cx: &RequestContext) -> Vec<String> {
  |          ^^^^^^^^^^^^^^^^^^^ the `?` operator cannot be applied to type `Vec<std::string::String>`
  |
  = help: the trait `Try` is not implemented for `Vec<std::string::String>`
